src/main/java/randoop/generation/ForwardGenerator.java

```diff
@@ -152,6 +187,27 @@ public ExecutableSequence step() {
     return eSeq;
   }
 
+  /**
+   * Updates eSeq's weight in weightMap according to the Orienteering formula from the GRT paper.
+   *
+   * @param eSeq the recently executed sequence which needs a weight update
+   */
+  private void setWeight(ExecutableSequence eSeq) {
+    SequenceExecDetails seqExecDets = sequenceExecutionCount.get(eSeq.sequence);
+    if (seqExecDets == null) {
+      seqExecDets = new SequenceExecDetails(0, 0);
+      sequenceExecutionCount.put(eSeq.sequence, seqExecDets);
+    }
+
+    // Increment the number of times this sequence has been executed.
+    seqExecDets.numExecutions += 1;
```

Comment:
Since this method is only ever called once, `numExecutions` is always 1.

The method needs to be called not just for the newly-created method, but also for each subsequence that makes up a sequence.  (This will require new logic.)  That is when updating will occur.
It would be hard to recompute the execution time of a sequence every time it is selected (its execution time is a fraction of the total execution time of the new sequence), and it is not likely to change much from execution to execution, so I think storing it once and assuming the same execution time thereafter is fine.